rpc: require integer verbosity; remove boolean 'verbose'

fqlx opened the pull request:
### Summary

Standardize RPC verbosity to integers only and remove boolean handling for clarity, consistency, and future extensibility.

### Rationale

 - **Legacy cleanup**: Boolean verbosity has been discouraged/deprecated in docs since 2017 and continues to create tech debt (special-case parsing, inconsistent tests, user confusion).
- **Consistency**: Integers enable multi-level output without overloading a boolean.

### User-visible changes
- getblock, getrawtransaction, getorphantxs no longer accept booleans.
- Passing true/false now errors with: Verbosity was boolean but only integer allowed.
- Migration: false → 0, true → 1.

### Code / Docs
- ParseVerbosity(arg, default) now rejects booleans (removed allow_bool).
- Call sites and functional tests updated to use integer levels. 
- Developer notes updated; release notes added.

**Breaking change:** scripts/tools using boolean verbosity must switch to integers.

stickies-v reviewed (commented):
Directionally, this is the right way to go, but I don't think we should be breaking the API for just this change. Let's do these cleanups when we can bundle them with other, necessary breaking changes, on a case-by-case basis. Approach NACK.

fqlx committed:
rpc: require integer verbosity; drop boolean 'verbose'

- Remove boolean handling from ParseVerbosity() and update callers.
- getblock/getrawtransaction/getrawmempool no longer accept true/false.
- Switch functional/Qt tests to integer verbosity.
- Update release notes and docs.

fqlx committed:
rpc-no-bool-verbosity: Add -deprecatedrpc=boolverbose flag

  1. rpc_deprecated.py: Tests that boolean verbosity fails without the deprecation flag
  2. rpc_boolverbose_deprecated.py: Tests that boolean verbosity works with the -deprecatedrpc=boolverbose flag
  3. Updated existing tests: Now expect the new "Boolean verbosity is deprecated" error message

fqlx commented on src/common/args.cpp:
```diff
@@ -900,3 +900,10 @@ std::pair<int, char**> WinCmdLineArgs::get()
 #endif
 } // namespace common
+
+bool IsDeprecatedRPCEnabled(const std::string& method)
```
Moved this back into `rpc` namespace `server.cpp`

maflcko reviewed (commented):
not sure where this pull is going, but it could make sense to split out the test changes into a separate pull (replacing the deprecated values with integers and named args), except for one deprecated value for each RPC, so that all code branches remain tests, but most code is using the new way and named args.
